### ISSUE TITLE
MD053 does not detect links within square brackets

### ISSUE THREAD

**Issue by MarkLodato** [DavidAnson/markdownlint#537]
MD053 incorrectly complains about link references that are contained within square brackets. This is perfectly fine Markdown syntax, rendered correctly and parsed by MD051 (see below), but MD053 fails to understand it.

## Reproducing

```markdown
# Test for MD053

MD053 [works] for regular reference, but not those [[within brackets]].

[works]: foo.html
[within brackets]: foo.html
```

```
$ npx markdownlint test.md
test.md:6:1 MD053/link-image-reference-definitions Link and image reference definitions should be needed [Unused link or image reference definition: "within brackets"] [Context: "[within brackets]: foo.html"]
```

## Further info

Note that MD051 does properly dereference the link and report it as broken (line 3) while simultaneously MD053 complains that the definition is unused (line 5).

```markdown
# Test for MD051

[[broken link]]

[broken link]: #broken
```

Result:

```
$ npx markdownlint test2.md
test2.md:3 MD051/link-fragments Link fragments should be valid [Context: "#broken"]
test2.md:5:1 MD053/link-image-reference-definitions Link and image reference definitions should be needed [Unused link or image reference definition: "broken link"] [Context: "[broken link]: #broken"]
```

**Comment by DavidAnson** [comment]
`MD051` uses the parser to scan valid links, whereas `MD053` uses custom code via `getReferenceLinkImageData`. This is why the "Further info" example behaves like that.

**Comment by ssbarnea** [comment]
This seems to be a regression introduced by `v0.32.1` release which affects people using markdown comments as [recommended here](https://stackoverflow.com/a/20885980/99834), which is hugely popular. We used the format below in order to add some placeholder comments which are needed when we compile the documentation, so we know were a specific section starts or ends.

```
[//]: # DO-NOT-REMOVE-README-TITLE
```

This worked fine until yesterday, when pre-commit monthly update upgraded markdownlint to v0.32.1.

Is there any workaround for this?

**Comment by nschonni** [comment]
@ssbarnea that's an odd hack, and isn't the same thing described in this issue

**Comment by ssbarnea** [comment]
I filed it as a regression at https://github.com/DavidAnson/markdownlint/issues/545

@nschonni Odd or not it with almost 2000 votes, it cannot be ignored.
